climada/engine/impact.py

```diff
@@ -1409,123 +1417,177 @@ def concat(cls, imp_list,concat_type='time_window'):
-            if len(exp_descriptions) > 1:
-               raise ValueError(f"The given impacts are based on exposures with different descriptions: {exp_descriptions}. "
-                            "The impacts are incompatible and cannot be concatenated.")      
+            check_exposure(imp_list)
+
+            #set attributes related to exposure
             coord_exp = imp_list[0].coord_exp
             tot_value = imp_list[0].tot_value
-          
+
             #fill remaining attributes
-            tag=imp_list[0].tag
+            tag = imp_list[0].tag
 
             #event IDs
             event_ids = [event_id for imp in imp_list for event_id in imp.event_id]
             if len(event_ids) != len(set(event_ids)):
-               raise ValueError("Duplicate event IDs found.")
+                raise ValueError("Duplicate event IDs found.")
             event_id = np.array(event_ids)
-          
+
             #event names
             event_names = [event_name for imp in imp_list for event_name in imp.event_name]
             if len(event_names) != len(set(event_names)):
-               raise ValueError("Duplicate event names found.")   
+                raise ValueError("Duplicate event names found.")
             event_name = event_names
 
             #impact matrix
             imp_mats =[imp.imp_mat for imp in imp_list]
             imp_mat = sparse.vstack(imp_mats)
-          
+
             #concatenate remaining attributes
             at_events = [imp.at_event for imp in imp_list]
             at_event = np.concatenate(at_events,axis=0)
             eai_exps = [imp.eai_exp for imp in imp_list]
-            eai_exp = np.nansum(eai_exps,axis=0) 
+            eai_exp = np.nansum(eai_exps,axis=0)
             aai_aggs = [imp.aai_agg for imp in imp_list]
             aai_agg = np.nansum(aai_aggs)
 
-        elif concat_type=='exposure':
-            
+        elif concat_type == 'exposure':
+
             ## some possibly useful code snippets for this option
             ##merge coordinates (only if exposure is not the same)
             ##check if there are any additional exposure coordinates in the new impact
             #compare_coords=np.isin(impact.coord_exp,self.coord_exp)
-            #additional_coords=[impact.coords[ind] for ind, x in enumerate(compare_coords) if not np.all(x)]
-            #imp.coord_exp=np.concatenate(self.coord_exp,additional_coords)
-        
-            raise ValueError("Concatenation of impacts with different exposures not yet implemented."
-                             "Feel free to do so!")
-     
-
-          
-
+            #ad_coords=[impact.coords[ind] for ind,x in enumerate(compare_coords) \
+            #                           if not np.all(x)]
+            #imp.coord_exp=np.concatenate(self.coord_exp,ad_coords)
 
+            raise ValueError("Concatenating impacts with different exposures not yet implemented."
+                             "Feel free to do so!")
```

Comment:
If not implemented, one can simply remove it. Or, then raise a NotImplementedError.

Reply:
Ok, I will remove it  that will simplify some other parts of the code as well